Implemented GUI

gamecoder95 opened the pull request:
- Implemented the GUI for the auction system and all the controls
- Integrated auction client with the GUI through the use of a callback
- Fixed the way the auction client was storing offers it made such that they are indexed by item number and not the request number.

At this point, the GUI is testable but not fully functional:
- If you run the test script, you can register with the GUI, but when attempting to send an offer, you offer denied saying that you're "not registered".

gamecoder95 committed:
Began the GUI

Created the main window with the registration panel.

gamecoder95 committed:
Added a label for the response

gamecoder95 committed:
Added deregister panel

Created deregister panel section and added it to the main window.

gamecoder95 committed:
Added new offer panel

Added the panel that allows user to create a new offer.

gamecoder95 committed:
Added gui for item list

Added the item list panel that will display the received items that
could be bid on.

gamecoder95 committed:
Added ongoing offers panel

Added the gui panel for displaying offers that are currently being bid
on.

gamecoder95 committed:
My offers panel

Made the "My Offers Panel" to contain the gui components for everything
related to what this client offers, including ongoing and ended offers.

gamecoder95 committed:
Completed basic GUI display

Changed the new item display so that it displays the information in a
column instead of a row.

gamecoder95 committed:
Removed hello world and added test package

gamecoder95 committed:
Added bid button for item section

gamecoder95 committed:
Fixed callbacks for buttons

Wrapped callbacks in "command" functions for each button so that they
may be called with certain parameters.

gamecoder95 committed:
Implemented GUI functions for register panel

When the user clicks the "Register" button, it invokes the callback and passes the input of the
entries to it as parameters.

gamecoder95 committed:
Began integration of gui and client

Updated client with the gui callback. Began implementing the gui
callback. Other gui updates like the helper class.

gamecoder95 committed:
Merge branch 'master' into gui

gamecoder95 committed:
Gui and client integration nearly done

samratdebroy commented on auctionsystem/client.py:
```diff
@@ -98,134 +104,124 @@ def handle_receive(self, data, addr=None):
+        # Handle UDP message to confirm registration for the client
         self.confirm_acknowledgement(req_num)
-        self.tcp_clients.clear()  # Should close all of the connections
-        self.udp_client.close_socket()
```
I think these two lines should not have been removed, prolly got lost during merge conflict resolution

gamecoder95 replied:
Good catch! Added them back in.

samratdebroy reviewed (changes requested):
Gui looks good, just need to readd those two missing lines in auctionsystem client that we due to merge conflicts

gamecoder95 committed:
Completed GUI features

Also restored restored the "getfqdn()" call in UDP server as it was
committed with it having been removed previously.

Completed all the remaining callbacks, implemented offer history, and
fixed the auction client so that the offers dictionary is indexed by
item number and not request number.

gamecoder95 committed:
Added back the lines in rcv dereg confirm to close the connections

samratdebroy reviewed (approved):
looks good
